spec.md

```diff
@@ -148,30 +148,33 @@ that contains both context and data).
 
 ### source
 * Type: Object
-* Description: This describes the software instance that emits the event at
-  runtime (i.e. the producer). It contains sub-properties (listed below)
+* Description: This describes the software where an event occurred. It contains
+  sub-properties (listed below)
 * Constraints:
   * REQUIRED
   * MUST contain at least one non-empty sub-property.
 
 ### source-type
 * Type: String
-* Description: Type of the event source. Providers define list of event
-  sources.
+* Description: The kind of object on which an event occurred. Providers
+  define list of source-types.
 * Constraints:
   * REQUIRED
   * MUST be a non-empty string
 * Examples:
   * s3
+  * google.firestore.document
 
 ### source-id
 * Type: String
-* Description: ID of the event source.
+* Description: A unique identifier of the object on which an event occurred. The
+  source-id is of kind source-type.
```

Comment:
I'm not sure what you mean by it "is of kind source-type". Does this mean it can't just be a UUID?

Reply:
Maybe the Google examples could help. In a google.storage.object.finalize event (when an object is ready to serve), the object is of type "google.storage.object" so source-ids must always be a path to an object, not a bucket, pubsub topic, etc.

Do you have suggestions to better describe this?

Reply:
That seems implementation specific since to some sources an ID could just be a uuid.
Would this wording work for you:
- - -
An identifier of the object on which the event occurred. The format
of this ID will be source specific.
- - -
And under Constraints, add:
- - -
MUST be unique within the scope of the `source`.
- - -
Then each source implementation can define the shape/syntax of this ID/string w/o forcing one model on everyone.

Reply:
"That seems implementation specific since to some sources an ID could just be a uuid." In some systems, a uuid would work, but I would say if and only if that UUID is used to serve data. I would have assumed that a PUT to a RESTful service at ${source.service}/${source.id} would generate an event with those respective fields.